### ISSUE TITLE
Window Focus on Click

### ISSUE THREAD

**Issue by JordanFitz** [Jebbs/DSFML#214]
After creating a basic window, if you alt+tab or click another window, the DSFML window obviously loses focus. The problem comes when you try to click back on the window. If you don't click on the window chrome itself, the window doesn't gain focus. This was a bug that was fixed in the original C++ version of SFML, as you can see [here](https://github.com/SFML/SFML/pull/457).

Details:
- Windows 10
- DMD32 v2.068.2
- DUB version 0.9.24
- DSFML 2.1.0
- DSFMLC 2.1

Sample code: 

```
import dsfml.graphics;

void main()
{
    auto window = new RenderWindow(VideoMode(640, 480), "Window");

    while(window.isOpen())
    {
        Event e;

        while(window.pollEvent(e))
        {
            if(e.type == Event.EventType.Closed)
            {
                window.close();
            }
        }       

        window.display();
    }
}
```

**Comment by aubade** [comment]
The solution to this will likely be pulled along when the underlying SFML version is updated; but I'm not sure what the status on that is

**Comment by Jebbs** [comment]
>but I'm not sure what the status on that is

Slow. :P

I've started working on it, but I have a lot of other things I'm trying to get in place as well. I don't have any kind of ETA at the moment.

**Comment by Jebbs** [comment]
I'll label this as fixed in the working branch as it is based on SFML 2.4, which fixed this issue.

Once it is merged into master I'll close it.

**Comment by Jebbs** [comment]
The current 2.4 work is now in master, which has SFML 2.4 as a dependency (where this issue has since been fixed).
